### PR TITLE
[Tidy] Clean-up examples where we set the id of grid and table within the function

### PR DESCRIPTION
**Pull request by maxschulz-COL**
## Description
Small task, where we eliminate all `id` setting within the grid or table functions.

## Notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

  - I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
  - I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorized to submit this contribution on behalf of the original creator(s) or their licensees.
  - I certify that the use of this contribution as authorized by the Apache 2.0 license does not violate the intellectual property rights of anyone else.
  - I have not referenced individuals, products or companies in any commits, directly or indirectly.
  - I have not added data or restricted code in any commits, directly or indirectly.

### CODE REVIEW

**Review by antonymilne — APPROVED** [review]
Thanks for doing this! 🙏

File: vizro-core/examples/demo/app.py [hunk 1]
```diff
@@ -438,7 +438,7 @@ def create_benchmark_analysis():
         components=[
             vm.AgGrid(
                 title="Click on a cell in country column:",
-                figure=dash_ag_grid(id="dash_ag_grid_country", data_frame=gapminder, columnDefs=columnsDefs),
+                figure=dash_ag_grid(data_frame=gapminder, columnDefs=columnsDefs),
```
Comment (huong-li-nguyen) [inline vizro-core/examples/demo/app.py:441]: Could you bring these changes over to vizro-internal as well? 🙏 To keep the deployed version in sync

**Review by huong-li-nguyen — APPROVED** [review]
Approved ✅

File: vizro-core/examples/demo/app.py (hunk 1 above)
Comment (maxschulz-COL) [reply vizro-core/examples/demo/app.py:441]: Will do, should I deploy it, or di we wait until a release?

**Pull request merged**